app/src/main/java/br/com/dionataferraz/vendas/account/AccountActivity.kt

```diff
@@ -27,30 +34,28 @@ class AccountActivity : AppCompatActivity() {
 
         configureActionBar()
         setContentView(binding.root)
+        configureSpinnerTypes()
         viewModel = AccountViewModel()
 
-        binding.btDeposit.setOnClickListener {
-            val value =
-                with(binding.valueET.text.toString()) { if (this.isBlank()) 0.0 else this.toDouble() }
-
-            viewModel.depositAcc(1, value)
-        }
-
-        binding.btRedeem.setOnClickListener {
+        binding.novaCompraBtnComprar.setOnClickListener {
+            getProgressBar().visibility = View.VISIBLE
             val value =
                 with(binding.valueET.text.toString()) { if (this.isBlank()) 0.0 else this.toDouble() }
+            val spinnerTypeValue = getSpinnerType().selectedItem.toString()
 
-            viewModel.withdrawAcc(1, value)
+            viewModel.registerItemBought(1, value, TransactionType.getEnumByValue(spinnerTypeValue))
         }
 
         viewModel.success.observe(this) { valorRetirada ->
             Toast.makeText(
                 this,
-                "saldo atualizado com sucesso",
+                "Compra realizada",
                 Toast.LENGTH_LONG
             ).show()
+
             val intent = Intent(this, HomeActivity::class.java)
             startActivity(intent)
+            getProgressBar().visibility = View.INVISIBLE
```

Comment:
Nesse caso com é uma progressbar nós podemos utilizar o GONE, pois ela não alguma ligação com a view
```suggestion
            getProgressBar().visibility = View.GONE
```